src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java

```diff
@@ -59,4 +61,12 @@ public void processQuestionnaireLinked(ResponseManagementEvent questionnaireLink
         questionnaireLinkedEvent.getEvent(),
         convertObjectToJson(uac));
   }
+
+  private void checkQidNotLinkedToAnotherCase(UacDTO uac, UacQidLink uacQidLink) {
+    if (uacQidLink.getCaze() != null) {
+      if (!uacQidLink.getCaze().getCaseId().equals(UUID.fromString(uac.getCaseId()))) {
+        throw new RuntimeException("UacQidLink already linked to a different case");
```

Comment:
Already linked to _which_ case? Put the case IDs in the exception so that we have something to go on.